common/src/main/java/com/sequenceiq/cloudbreak/logger/RestLoggerFilter.java

```diff
@@ -0,0 +1,90 @@
+package com.sequenceiq.cloudbreak.logger;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.util.ContentCachingRequestWrapper;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+
+import com.sequenceiq.cloudbreak.common.anonymizer.AnonymizerUtil;
+
+public class RestLoggerFilter extends OncePerRequestFilter {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final DateFormat formatter;
+
+    private final boolean restLoggerEnabled;
+
+    public RestLoggerFilter(boolean restLoggerEnabled) {
+        this.restLoggerEnabled = restLoggerEnabled;
+        this.formatter = new SimpleDateFormat("HH:mm:ss.SSS");
+        this.formatter.setTimeZone(TimeZone.getTimeZone("UTC"));
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+        Date start = new Date(System.currentTimeMillis());
+        ContentCachingRequestWrapper wrappedRequest = getWrappedRequest(request);
+        ContentCachingResponseWrapper wrappedResponse = getWrappedResponse(response);
+
+        filterChain.doFilter(wrappedRequest, wrappedResponse);
+
+        if (restLoggerEnabled) {
+            Date end = new Date(System.currentTimeMillis());
+            String log = new StringBuilder()
```

Comment:
Log should be pushed through anonymization filter

Reply:
the request and response pairs are already anonymized do we all other field as well ?

Reply:
Oh, I haven't seen it, so I guess then it should be fine.